addons/sale_coupon/security/ir.model.access.csv

```diff
@@ -7,3 +7,5 @@ access_coupon_salesman,salesman,model_sale_coupon,sales_team.group_sale_salesman
 access_coupon_manager,program manager,model_sale_coupon,sales_team.group_sale_manager,1,1,1,0
 access_reward_salesman,salesman,model_sale_coupon_reward,sales_team.group_sale_salesman,1,0,0,0
 access_reward_manager,program manager,model_sale_coupon_reward,sales_team.group_sale_manager,1,1,1,0
+"access_sale_coupon_apply_code","access.sale.coupon.apply.code","model_sale_coupon_apply_code","base.group_system",1,0,1,0
+"access_sale_coupon_generate","access.sale.coupon.generate","model_sale_coupon_generate","base.group_system",1,0,1,0

```

Comment:
both these ACLs should be applied on the `sales_team.group_sale_salesman` instead, Settings is too restrictive to apply coupons on orders

Reply:
i would also add a 'your own wizards only' rule by default

Reply:
By the way, sale.coupon.generate uses a `safe_eval` that should be replaced by a `ast.literal_eval` (cf #43463)

Reply:
i see you've done it already, i was about to do it now. thx!

Reply:
yep, realised there were a lot of those ;-)